src/components/structures/MatrixChat.js

```diff
@@ -589,7 +589,16 @@ module.exports = React.createClass({
                     var theAlias = MatrixTools.getCanonicalAliasForRoom(room);
                     if (theAlias) presentedId = theAlias;
                 }
-                self.notifyNewScreen('room/'+presentedId);
+
+                if (typeof presentedId != 'undefined') {
```

Comment:
We generally haven't used typeof to check undefined here (and in this case It's unnecessary since we know for sure it's defined). I would probably just drop the `typeof`.